### PR TITLE
tweak(tupaiaWeb): RN-1383: Make raw data downloads use one time link

### PR DESCRIPTION
**Pull request by alexd-bes**
### Changes:
- Updated `constructExportEmail` to handle links to different platforms
- Removed option for emailing attached data
- Moved `getTempDirectory` and `getExportPathForUser` to `server-utils`
- Make `ExportSurveyDataHandler` in `web-config-server` use same temp directory as everywhere else

### CODE REVIEW

File: packages/server-utils/src/constructExportEmail.ts [hunk 1]
```diff
@@ -48,18 +37,36 @@ ${error}`,
...
-    };
+  const platformKey = platform || 'adminPanel';
+
+  /**
+   * We don't yet have single sign-on across all platforms, so we need to know which platform the user is on so the correct session is used, from where they requested the export. We also can't use central-server api url for this directly because there needs to be an auth header in the request.
+   */
+  const PLATFORM_SETTINGS = {
+    tupaia: {
+      url: requireEnv('TUPAIA_WEB_SERVER_API_URL'),
+      friendlyName: 'Tupaia',
```
Comment (tcaiger) [inline packages/server-utils/src/constructExportEmail.ts:48]: I like the term friendlyName :)

File: packages/server-utils/src/constructExportEmail.ts [hunk 2]
```diff
@@ -28,13 +17,13 @@ type ResponseBody = {
...
-    emailExportFileMode?: EmailExportFileModes;
+    platform?: 'tupaia' | 'adminPanel' | 'datatrak';
   };
 };
 
 export const constructExportEmail = async (responseBody: ResponseBody, req: Req) => {
-  const { emailExportFileMode = EmailExportFileModes.DOWNLOAD_LINK } = req.query;
   const { error, filePath } = responseBody;
+  const { platform } = req.query;
   const subject = 'Your export from Tupaia';
```
Comment (tcaiger) [inline packages/server-utils/src/constructExportEmail.ts:27]: I wonder if the subject should change depending on the platform?

File: packages/server-utils/src/constructExportEmail.ts [hunk 3]
```diff
@@ -28,13 +17,13 @@ type ResponseBody = {
 
 type Req = {
   query: {
-    emailExportFileMode?: EmailExportFileModes;
+    platform?: 'tupaia' | 'adminPanel' | 'datatrak';
```
Comment (tcaiger) [inline packages/server-utils/src/constructExportEmail.ts:20]: Should Lesmis be included as an option?

File: packages/server-utils/src/constructExportEmail.ts (hunk 2 above)
Comment (alexd-bes) [reply packages/server-utils/src/constructExportEmail.ts:27]: Yes, good idea

File: packages/server-utils/src/constructExportEmail.ts (hunk 3 above)
Comment (alexd-bes) [reply packages/server-utils/src/constructExportEmail.ts:20]: Good idea

**Comment by alexd-bes** [comment]
@tcaiger are you able to re-review this one?

**Review by tcaiger — APPROVED** [review]

**Pull request merged**